### PR TITLE
update deps and added test

### PR DESCRIPTION
**Pull request by aricart**
updated dev dependencies
added test to verify that on cert error, client redials

### CODE REVIEW

File: lib/nats.js [hunk 1]
```diff
@@ -886,6 +886,7 @@ Client.prototype.sendConnect = function () {
  * @api private
  */
 Client.prototype.createConnection = function () {
+  this.emit('dialing')
```
Comment (derekcollison) [inline lib/nats.js:889]: We freeform these all the time or are they constants somewhere?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "nats",
-  "version": "1.4.9-0",
+  "version": "1.4.9-2",
```
Comment (derekcollison) [inline package.json:3]: The -N for betas?

File: lib/nats.js (hunk 1 above)
Comment (aricart) [reply lib/nats.js:889]: these are only for testing - we document all the ones that are official configuration/event properties.

File: package.json (hunk 2 above)
Comment (aricart) [reply package.json:3]: yes

File: lib/nats.js (hunk 1 above)
Comment (derekcollison) [reply lib/nats.js:889]: But these will be in production too and users can start to depend on them yes?

Comment (aricart) [reply lib/nats.js:889]: we are not responsible for users digging into the source and making dependencies on non-public features. This is simply used by a test to confirm that a dial took place when it is aborted by a tls certificate error.

Comment (derekcollison) [reply lib/nats.js:889]: Understood but they can, lets just make it a constant. Don't have to document it.

Comment (kozlovic) [reply lib/nats.js:889]: Would there be a way to use existing connection field that are updated in this process, for instance `reconnects` (I understand that this test fails during initial connect, but wonder if node is using the same logic than if it was reconnecting even on initial failed connect).

Comment (aricart) [reply lib/nats.js:889]: yes hooking on `reconnecting` works perfect. - will add constants for events in a different PR. this way we can group them under `Events` and `TestEvents` to further document the intention..

**Pull request merged**